deploy/stacks/container.py

```diff
@@ -163,129 +147,161 @@ def __init__(
-            container_id=f'container',
-            ecr_repository=ecr_repository,
-            environment=self._create_env('DEBUG'),
-            image_tag=cdkproxy_image_tag,
-            log_group=self.create_log_group(
-                envname, resource_prefix, log_group_name='policies-updater'
-            ),
-            schedule_expression=Schedule.expression('rate(15 minutes)'),
-            scheduled_task_id=f'{resource_prefix}-{envname}-policies-updater-schedule',
-            task_id=f'{resource_prefix}-{envname}-policies-updater',
-            task_role=self.task_role,
-            vpc=vpc,
-            security_group=scheduled_tasks_sg,
-            prod_sizing=prod_sizing,
-        )
-        self.ecs_security_groups.extend(
-            update_bucket_policies_task.task.security_groups
+        ssm.StringParameter(
+            self,
+            f'ECSClusterNameParam{envname}',
+            parameter_name=f'/dataall/{envname}/ecs/cluster/name',
+            string_value=cluster.cluster_name,
         )
 
-        # TODO introduce the ability to change the deployment depending on config.json file
-        subscriptions_task, subscription_task_def = self.set_scheduled_task(
-            cluster=cluster,
-            command=[
-                'python3.8',
-                '-m',
-                'dataall.modules.datasets.tasks.dataset_subscription_task',
-            ],
-            container_id=f'container',
-            ecr_repository=ecr_repository,
-            environment=self._create_env('INFO'),
-            image_tag=cdkproxy_image_tag,
-            log_group=self.create_log_group(
-                envname, resource_prefix, log_group_name='subscriptions'
+        ssm.StringParameter(
+            self,
+            f'VPCPrivateSubnetsParam{envname}',
+            parameter_name=f'/dataall/{envname}/ecs/private_subnets',
+            string_value=','.join(
+                vpc.select_subnets(
+                    subnet_type=ec2.SubnetType.PRIVATE_WITH_NAT
+                ).subnet_ids
             ),
-            schedule_expression=Schedule.expression('rate(15 minutes)'),
-            scheduled_task_id=f'{resource_prefix}-{envname}-subscriptions-schedule',
-            task_id=f'{resource_prefix}-{envname}-subscriptions',
-            task_role=self.task_role,
-            vpc=vpc,
-            security_group=scheduled_tasks_sg,
-            prod_sizing=prod_sizing,
         )
-        self.ecs_security_groups.extend(subscriptions_task.task.security_groups)
 
+        self.ecs_cluster = cluster
+
+        self.ecs_task_definitions = [
+            cdkproxy_task_definition,
+            catalog_indexer_task.task_definition,
+        ]
+
+        self.add_sync_dataset_table_task()
+        self.add_bucket_policy_updater_task()
+        self.add_subscription_task()
+        self.add_share_management_task()
+
+    @run_if("modules.datasets.active")
```

Comment:
Is it expected that we have further module-dependent ECS tasks in the near future?
If the answer is yes, I'd think about generalizing further your solution:
1. keeping a mapping between: modules -> module related ECS tasks (eg. in config.json)
2. iterating through the active modules (maybe taken/saved from the module loader?) and spawning its ECS tasks here

Reply:
I believe at this point, no one can say for sure whether we have a new ECS tasks or not.
As of today, only datasets are using deployment changes and we can't predict how things will evolve in the future. So, we should stay as flexible as we can :) 

Plus, the potential changes can be everywhere and not only related to modularization. Some can be small (say like existing  `enable_cw_canaries`) and some big ( `opensearch serverless`). Of course, if we decide not to use `cdk.json` for configuring those properties